netsim/sniffer/process.py

```diff
@@ -0,0 +1,402 @@
+        d = json.load(f)
+        f.close()
+        return d
+    return None
+
+def load_pcap(path, keylog=None, topo_path=None):
+    parameters_dict = {}
+    if keylog:
+        parameters_dict = {'-o' : "ssl.keylog_file:" + keylog}
+    cap_json = pyshark.FileCapture(path, custom_parameters=parameters_dict)
+    packets = []
+    str_packet_data = []
+
+    topo = read_topo(topo_path)
+    topo_nodes = []
+    topo_node_ip_map = {}
+    if topo:
+        i=0
+        for node in topo['nodes']:
+            if node['type'] == 'Host' or node['type'] == 'NAT':
+                topo_nodes.append(node)
+                topo_node_ip_map[node['ip']] = i
+                i+=1
+
+    node_list = []
+    ip_set = set()
+
+    for packet in cap_json:
+        if len(packet.layers) == 2:
+            if 'arp' in packet: # ignore ARP
+                continue
+        if 'ip' in packet:
+            ip_set.add(packet.ip.src)
+            ip_set.add(packet.ip.dst)
+            p = {
+                'src': packet.ip.src,
+                'dst': packet.ip.dst,
+                'type': '',
+                'ipv6': False
+            }           
+
+            if 'tcp' in packet:
+                p['type'] = 'TCP'
+            elif 'udp' in packet:
+                p['type'] = 'UDP'
+
+            if 'quic' in packet:
+                p['type'] = 'QUIC'
+            elif 'http' in packet:
+                p['type'] = 'HTTP'
+            elif 'stun' in packet:
+                p['type'] = 'STUN'
+            elif 'icmp' in packet:
+                p['type'] = 'ICMP'
+            
+            packets.append(p)
+            zk = packet.__str__().replace('\n', '<\\n>')
+            str_packet_data.append(zk)
+
+        if 'ipv6' in packet:
+            p = {
+                'src': packet.ipv6.src,
+                'dst': packet.ipv6.dst,
+                'type': 'ICMPv6',
+                'ipv6': True
+            }
+
+    js_packet_data = "const pkt_data = ["
+    for i in range(len(str_packet_data)):
+        js_packet_data += '`{}`,'.format(str_packet_data[i])
+    js_packet_data += "];"
+
+    for ip in ip_set:
+        n = {
+            'type': 'node',
+            'id': ip,
+            'ip': ip
+        }
+        if ip in topo_node_ip_map:
+            tn = topo_nodes[topo_node_ip_map[ip]]
+            if tn['name'].startswith('1_relay'):
+                n['type'] = 'relay'
+                n['id'] = tn['name']
+            if tn['name'].startswith('iroh'):
+                n['type'] = 'iroh'
+                n['id'] = tn['name'].replace('iroh_', '')
+            if tn['name'].startswith('i_'):
+                n['type'] = 'iroh'
+                n['id'] = tn['name'].replace('i_', '')
+            elif tn['name'].startswith('n_'):
+                n['type'] = 'nat'
+                n['id'] = tn['name']
+            
+        node_list.append(n)
+    return packets, js_packet_data, node_list
+
+class NetsimViz():
+
+    HOVER_JS = """
```

Comment:
Maybe write this in a `.js` file next to this and find it using `__file__`?

(Yes, using `__file__` for this is frowned upon but I don't think this stuff is a proper python distribution that can use the right way to load other resources.  And that's already a bit easier and doesn't force you into python packaging world)